### PR TITLE
Implement docker multi stage build and use common Go image

### PR DESCRIPTION
**Pull request by chimanjain**
# Description
Implement docker multi stage build to build module and use common Go image for building module.
Also update github actions.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| https://github.com/dell/csm/issues/1098 |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Build module image and ran sanity test
 
Screenshot 2024-02-19 194001

### CODE REVIEW

**Review by HarishH-DELL — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -37,13 +37,7 @@ K3S_SELINUX_VERSION ?= 0.4-1
 .PHONY: build
 build:
 	-mkdir -p ./bin
-	cp Dockerfile ./bin/Dockerfile
-	CGO_ENABLED=0 go build -o ./bin ./cmd/proxy-server/
 	CGO_ENABLED=0 go build -o ./bin ./cmd/karavictl/
-	CGO_ENABLED=0 go build -o ./bin ./cmd/sidecar-proxy/
-	CGO_ENABLED=0 go build -o ./bin ./cmd/tenant-service/
-	CGO_ENABLED=0 go build -o ./bin ./cmd/role-service/
-	CGO_ENABLED=0 go build -o ./bin ./cmd/storage-service/
```
Comment (shaynafinocchiaro) [inline Makefile:46]: Why are build targets for the services being removed?

File: Makefile [hunk 2]
```diff
@@ -86,11 +80,11 @@ redeploy: build builder
 .PHONY: builder
 builder: build download-csm-common
 	$(eval include csm-common.mk)
-	$(BUILDER) build -t localhost/proxy-server:$(BUILDER_TAG) --build-arg APP=proxy-server --build-arg BASEIMAGE=$(DEFAULT_BASEIMAGE) ./bin/.
-	$(BUILDER) build -t localhost/sidecar-proxy:$(SIDECAR_TAG) --build-arg APP=sidecar-proxy --build-arg BASEIMAGE=$(DEFAULT_BASEIMAGE) ./bin/.
-	$(BUILDER) build -t localhost/tenant-service:$(BUILDER_TAG) --build-arg APP=tenant-service --build-arg BASEIMAGE=$(DEFAULT_BASEIMAGE) ./bin/.
-	$(BUILDER) build -t localhost/role-service:$(BUILDER_TAG) --build-arg APP=role-service --build-arg BASEIMAGE=$(DEFAULT_BASEIMAGE) ./bin/.
-	$(BUILDER) build -t localhost/storage-service:$(BUILDER_TAG) --build-arg APP=storage-service --build-arg BASEIMAGE=$(DEFAULT_BASEIMAGE) ./bin/.
+	$(BUILDER) build -t localhost/proxy-server:$(BUILDER_TAG) --build-arg APP=proxy-server --build-arg GOIMAGE=$(DEFAULT_GOIMAGE) --build-arg BASEIMAGE=$(DEFAULT_BASEIMAGE) .
```
Comment (atye) [inline Makefile:83]: Is `DEFAULT_GOIMAGE` defined elsewhere?

File: Makefile (hunk 1 above)
Comment (atye) [reply Makefile:46]: The service gets built inside the Golang image and then copied into the final image.

https://github.com/dell/karavi-authorization/blob/50872f0647daae010002adb1a65d4e04ee7a45a4/Dockerfile#L26
https://github.com/dell/karavi-authorization/blob/50872f0647daae010002adb1a65d4e04ee7a45a4/Dockerfile#L38

**Review by shaynafinocchiaro — APPROVED** [review]

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by atye** [comment]
run e2e test

**Comment by tdawe** [comment]
e2e pass

**Comment by atye** [comment]
e2e pass

**Pull request merged**